.pre-commit-config.yaml

```diff
@@ -0,0 +1,51 @@
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: check-executables-have-shebangs
+    -   id: check-merge-conflict
+    -   id: debug-statements
+    -   id: check-yaml
+        files: .*\.(yaml|yml)$
+
+-   repo: https://github.com/asottile/reorder_python_imports
+    rev: v2.6.0
+    hooks:
+    -   id: reorder-python-imports
+        files: ^src/|tests/
+
+-   repo: local
+    hooks:
+    -   id: mypy
+        name: mypy
+        entry: mypy src
+        language: system
+        pass_filenames: false
+
+# Black: thanks, I hate it.
+# -   repo: local
+#     hooks:
+#     -   id: black
+#         name: black
+#         entry: black
+#         files: ^src/|tests/
+#         language: system
```

Comment:
why aren't we using this?

Reply:
Hi @MJJoyce : Yeah, the `exclude_types` worked only for `.xml` files. If I set
```
exclude_types: [xml, tab, TAB]
```
then Black complains of an `unknown file type tab`. If I set
```
exclude_types: [xml]
```
then it left the XML files alone but clobbered the `.tab` and `.TAB` files. So what goes in `exclude_types` is not a file extension but some kind of a file definition. I didn't feel it was worthwhile pursuing how you "define new file types" to Black, especially since it seemed to involve writing an extension module.

If I moved the test data symlink out and put tests into their own top level directory, then I could exclude the entire directory from Black by virtue of just giving `src/` as the place to look. But then `pytest` failed to find any tests. And by that point I felt I was spending way too much time on this, particularly since I had over 250 different flake8 issues to rectify.